### DISCUSSION TITLE
umi首次启动时的加载动画是怎么实现的

### DISCUSSION THREAD

**Discussion by flycran** [umijs/umi#12375] [Q&A]
我对umi首次启动时的加载动画很感兴趣，很多中后台框架都有这个功能，但是我居然在网上找不到一个文章来系统地讲述怎么实现这个功能。能不能出一篇文章讲一讲这个效果是怎么实现的，我相信很多人也都对这个效果感兴趣。或者有没有现有的文章专门讲这个效果如何实现的。

**Comment by fz6m** [comment]
开发的时候有个 express 来提供服务，用 webpack 的 progress 插件监听下什么时候编译完，同时也就知道进度了，在没编译完之前，express 服务提供一个加载动画的 html 即可，里面的进度通过不断轮询来更新，等编译好了，就不渲染这个加载的 html 了。
